db/main.py

```diff
@@ -0,0 +1,68 @@
+from fastapi import FastAPI, Depends, HTTPException
+from sqlalchemy.orm import Session
+
+import crud
+import schemas
+from db.database import SessionLocal
+
+app = FastAPI()
+
+
+def get_db() -> Session:
+    db = SessionLocal()
+
+    try:
+        yield db
+    finally:
+        db.close()
+
+
+@app.get("/")
+def root() -> dict:
+    return {"message": "Hello from Library"}
+
+
+@app.get("/books/", response_model=list[schemas.Book])
+def read_books(db: Session = Depends(get_db)):
+    return crud.get_all_books(db=db)
+
+
+@app.post("/books/", response_model=schemas.Book)
+def create_book(
+    book: schemas.BookCreate,
+    db: Session = Depends(get_db),
+):
+    db_book = crud.get_book_by_title(db=db, title=book.title)
+
+    if db_book:
+        raise HTTPException(
+            status_code=400,
+            detail="Such title for Book already exists"
+        )
+
+    return crud.create_book(db=db, book=book)
+
+
+@app.get("/authors/", response_model=list[schemas.Author])
+def read_authors(
+    book: str | None = None,
+    db: Session = Depends(get_db),
+):
+    return crud.get_author_list(
+        db=db, books=book
+    )
+
+
+@app.get("/authors/{author_id}/", response_model=schemas.Author)
+def read_single_author(author_id: int, db: Session = Depends(get_db)):
+    db_author = crud.get_author(db=db, author_id=author_id)
+
+    if db_author is None:
+        raise HTTPException(status_code=404, detail="Author not found")
+
+    return db_author
+
+
+@app.post("/authors/", response_model=schemas.Author)
+def create_author(author: schemas.AuthorCreate, db: Session = Depends(get_db)):
+    return crud.create_author(db=db, author=author)
```

Comment:
The `create_author` endpoint uses `schemas.AuthorCreate`, which includes a `book_id` field. This might not be necessary or correct given the one-to-many relationship between authors and books. Typically, the `book_id` would be part of the `Book` model, referencing the `Author`. Consider reviewing the relationship setup in your models to ensure consistency.